lib/auth/register.go

```diff
@@ -740,23 +756,89 @@ func registerUsingAzureMethod(
 		}
 
 		return &proto.RegisterUsingAzureMethodRequest{
-			RegisterUsingTokenRequest: &types.RegisterUsingTokenRequest{
-				Token:                token,
-				HostID:               params.ID.HostUUID,
-				NodeName:             params.ID.NodeName,
-				Role:                 params.ID.Role,
-				AdditionalPrincipals: params.AdditionalPrincipals,
-				DNSNames:             params.DNSNames,
-				PublicTLSKey:         params.PublicTLSKey,
-				PublicSSHKey:         params.PublicSSHKey,
-			},
-			AttestedData: ad,
-			AccessToken:  accessToken,
+			RegisterUsingTokenRequest: registerUsingTokenRequestForParams(token, params),
+			AttestedData:              ad,
+			AccessToken:               accessToken,
 		}, nil
 	})
 	return certs, trace.Wrap(err)
 }
 
+// registerUsingTPMMethod is used to register using the TPM join method. It
+// is able to register through a proxy or through the auth server directly.
+func registerUsingTPMMethod(
+	ctx context.Context,
+	client joinServiceClient,
+	token string,
+	params RegisterParams,
+) (*proto.Certs, error) {
+	log := slog.Default()
```

Comment:
I don't think you want the default logger, afaik we use a custom handler to make it look like the logrus logs.

Reply:
We set the Default logger to this customized logger during initialisation - so no worries here. I didn't want to bubble this up to the Register arg yet since slog usage is still pretty rare. Once Machine ID is converted, I'll see about converting all the registration stuff.

Reply:
> We set the Default logger to this customized logger during initialisation

Oh I didn't know that. Nice.